indexer/src/transform.test.ts

```diff
@@ -0,0 +1,604 @@
+  if ("receipt" in result[1].data) {
+    assertEquals(result[1].data.receipt, ethReceipt);
+  } else throw new Error("Expected receipt not found in result data");
+  if ("header" in result[2].data) {
+    assertEquals(result[2].data.header.number, expectedHeader.number);
+    assertEquals(result[2].data.header.hash, expectedHeader.hash);
+    assertEquals(result[2].data.header.parentHash, expectedHeader.parentHash);
+    assertEquals(result[2].data.header.sha3Uncles, expectedHeader.sha3Uncles);
+    assertEquals(result[2].data.header.stateRoot, expectedHeader.stateRoot);
+    assertEquals(result[2].data.header.gasLimit, expectedHeader.gasLimit);
+    assertEquals(
+      result[2].data.header.withdrawalsRoot,
+      expectedHeader.withdrawalsRoot,
+    );
+    assertEquals(result[2].data.header.transactions.length, 0);
+  } else {
+    throw new Error("Expected header not found in result data");
+  }
+});
+
+Deno.test("transform with logs events and trasaction", async () => {
+  const { result, expectedHeader, ethTx, ethLogs, ethReceipt } =
+    await transformData(
+      header,
+      events,
+      transactions,
+      transaction,
+      receipt,
+      event,
+    );
+
+  assertEquals(result.length, 4);
+  if ("tx" in result[0].data) assertEquals(result[0].data.tx, ethTx);
+  else throw new Error("Expected tx not found in result data");
+  if ("receipt" in result[1].data) {
+    assertEquals(result[1].data.receipt, ethReceipt);
+  } else throw new Error("Expected receipt not found in result data");
+  if ("log" in result[2].data) assertEquals(result[2].data.log, ethLogs[0]);
+  else throw new Error("Expected log not found in result data");
+  if ("header" in result[3].data) {
+    assertEquals(result[3].data.header.number, expectedHeader.number);
+    assertEquals(result[3].data.header.hash, expectedHeader.hash);
+    assertEquals(result[3].data.header.parentHash, expectedHeader.parentHash);
+    assertEquals(result[3].data.header.sha3Uncles, expectedHeader.sha3Uncles);
+    assertEquals(result[3].data.header.stateRoot, expectedHeader.stateRoot);
+    assertEquals(result[3].data.header.gasLimit, expectedHeader.gasLimit);
+    assertEquals(
+      result[3].data.header.withdrawalsRoot,
+      expectedHeader.withdrawalsRoot,
+    );
+    assertEquals(result[3].data.header.transactions.length, 0);
+  } else {
+    throw new Error("Expected header not found in result data");
+  }
+});
+
+Deno.test("transform with real data", async () => {
+  const headersList = transactionsData.headersList;
+  const eventsList = transactionsData.eventsList;
+  const transactionsList = transactionsData.transactionsList;
+
+  for (let i = 0; i < headersList.length; i++) {
+    const header = headersList[i];
+    const events = eventsList[i];
+    const transactions = transactionsList[i];
+
+    const result = await transform({
+      header: header,
+      events: events,
+      transactions: transactions,
+    });
+
+    assertExists(result);
```

Comment:
let's maybe make an issue to update this, I think `assertExists` is not a strong enough assertion on the expected output

Reply:
https://github.com/kkrt-labs/kakarot-rpc/issues/1258